src/managers/KeybindManager.cpp

```diff
@@ -1820,6 +1821,28 @@ void CKeybindManager::focusWindow(std::string regexp) {
     g_pCompositor->warpCursorTo(PWINDOW->middle());
 }
 
+void CKeybindManager::toggleSwallow(std::string args) {
+    CWindow* pWindow = g_pCompositor->m_pLastWindow;
+    if (pWindow->m_pSwallowed && g_pCompositor->windowExists(pWindow->m_pSwallowed)) {
```

Comment:
windowExists should be avoided in favor of methods that don't leave dangling pointers, like e.g. removing them in ::onUnmap

Reply:
I'm not sure what to replace it with. I used the same condition found in `src/events/windows.cpp:690`.
Should I remove the check altogether ?
Is it safe to do so and assume the pointer isn't dangling ? `if (pWindow->m_pSwallowed) {`

Reply:
I think it should be valid if you add a method to remove the `swallowed` ptr in `::onUnmap` (basically when a window is unmapped, loop through all windows and set swallowed to null wherever it finds a ref)

Reply:
sure